tap_github/streams.py

```diff
@@ -938,10 +939,95 @@ class ContributorsStream(GitHubStream):
+    ) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        assert context is not None, f"Context cannot be empty for '{self.name}' stream."
+        params = super().get_url_params(context, next_page_token)
+        # Fetch all contributions, including anonymous users.
+        params["anon"] = "true"
+        return params
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of anonymous contributors."""
+        parsed_response = super().parse_response(response)
+        return filter(lambda x: x["type"] == "Anonymous", parsed_response)
+
+    schema = th.PropertiesList(
+        # Parent keys
+        th.Property("repo", th.StringType),
+        th.Property("org", th.StringType),
+        # Anonymous contributor keys
         th.Property("email", th.StringType),
         th.Property("name", th.StringType),
-        # Shared keys
         th.Property("type", th.StringType),
         th.Property("contributions", th.IntegerType),
     ).to_dict()
+
+
+class StargazersStream(GitHubStream):
+    """Defines 'Stargazers' stream. Warning: this stream does NOT track star deletions."""
+
+    name = "stargazers"
+    path = "/repos/{org}/{repo}/stargazers"
+    primary_keys = ["repo", "org", "user_id"]
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    replication_key = "starred_at"
+
+    @property
+    def http_headers(self) -> dict:
+        """Return the http headers needed.
+
+        Overridden to use an endpoint which includes starred_at property:
+        https://docs.github.com/en/rest/reference/activity#custom-media-types-for-starring
+        """
+        headers = super().http_headers
+        headers["Accept"] = "application/vnd.github.v3.star+json"
+        return headers
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        """
+        Add a user_id top-level field to be used as state replication key.
+        """
+        row["user_id"] = row["user"]["id"]
+        return row
+
+    schema = th.PropertiesList(
+        # Parent Keys
+        th.Property("repo", th.StringType),
+        th.Property("org", th.StringType),
+        th.Property("user_id", th.StringType),
+        # Stargazer Info
+        th.Property("starred_at", th.DateTimeType),
+        th.Property(
+            "user",
+            th.ObjectType(
+                th.Property("login", th.StringType),
+                th.Property("id", th.IntegerType),
+                th.Property("node_id", th.StringType),
+                th.Property("avatar_url", th.StringType),
+                th.Property("gravatar_id", th.StringType),
+                th.Property("html_url", th.StringType),
+                th.Property("type", th.StringType),
+                th.Property("site_admin", th.BooleanType),
+            ),
+        ),
+    ).to_dict()
```

Comment:
This looks like a spillover from another PR, no?

Reply:
Not sure on which commit # you got this 🤷 , probably a merge of main?